### ISSUE TITLE
Add an .event_dim property to Distributions

### ISSUE THREAD

**Issue by fritzo** [pyro-ppl/pyro#273]
This is useful for determining the number of dimensions in parameters of a `Distribution` object. See also #153 . We may also want an `.event_size` with full shape information.

For most distributions `.event_dim == 1`. For univariate distributions, `event_size == (1,)`. For a matrix distribution like `Wishart` `.event_dim == 2` and `.event_size == (n,n)` for some `n`.

Note that the `_dim` and `_size` match Pytorch's `.dim()` and `.size()` methods. If Pytorch moves towards Numpy/Tensorflow's `.shape` we can switch to `_shape` like Edward.

(edited to use `event_dim` instead of `param_dim`)
